systems/data-plan/api-gateway/pkg/rest/router.go

```diff
@@ -0,0 +1,256 @@
+	pb "github.com/ukama/ukama/systems/data-plan/package/pb"
+	"github.com/wI2L/fizz"
+	"github.com/wI2L/fizz/openapi"
+)
+
+const ORG_ID = 12345
+
+type Router struct {
+	f       *fizz.Fizz
+	clients *Clients
+	config  *RouterConfig
+}
+
+type RouterConfig struct {
+	metricsConfig config.Metrics
+	httpEndpoints *pkg.HttpEndpoints
+	debugMode     bool
+	serverConf    *rest.HttpConfig
+}
+
+type Clients struct {
+	d dataPlan
+}
+
+type dataPlan interface {
+	AddPackage(req *pb.AddPackageRequest) (*pb.AddPackageResponse, error)
+	UpdatePackage(req *pb.UpdatePackageRequest) (*pb.UpdatePackageResponse, error)
+	GetPackage(req *pb.GetPackagesRequest) (*pb.GetPackagesResponse, error)
+	DeletePackage(req *pb.DeletePackageRequest) (*pb.DeletePackageResponse, error)
+	UploadBaseRates(req *pbBaseRate.UploadBaseRatesRequest) (*pbBaseRate.UploadBaseRatesResponse, error)
+	GetBaseRates(req *pbBaseRate.GetBaseRatesRequest) (*pbBaseRate.GetBaseRatesResponse, error)
+	GetBaseRate(req *pbBaseRate.GetBaseRateRequest) (*pbBaseRate.GetBaseRateResponse, error)
+}
+
+func NewClientsSet(endpoints *pkg.GrpcEndpoints) *Clients {
+	c := &Clients{}
+	c.d = client.NewDataPlan(endpoints.Package, endpoints.BaseRate, endpoints.Timeout)
+
+	return c
+}
+
+func NewRouter(clients *Clients, config *RouterConfig) *Router {
+
+	r := &Router{
+		clients: clients,
+		config:  config,
+	}
+
+	if !config.debugMode {
+		gin.SetMode(gin.ReleaseMode)
+	}
+
+	r.init()
+	return r
+}
+
+func NewRouterConfig(svcConf *pkg.Config) *RouterConfig {
+	return &RouterConfig{
+		metricsConfig: svcConf.Metrics,
+		httpEndpoints: &svcConf.HttpServices,
+		serverConf:    &svcConf.Server,
+		debugMode:     svcConf.DebugMode,
+	}
+}
+
+func (rt *Router) Run() {
+	logrus.Info("Listening on port ", rt.config.serverConf.Port)
+	err := rt.f.Engine().Run(fmt.Sprint(":", rt.config.serverConf.Port))
+	if err != nil {
+		logrus.Error(err)
+	}
+}
+
+func (r *Router) init() {
+	r.f = rest.NewFizzRouter(r.config.serverConf, pkg.SystemName, version.Version, r.config.debugMode)
+	v1 := r.f.Group("/v1", "Data-plan system ", "Data-plan  system version v1")
+
+	baseRates := v1.Group("/baseRates", "BaseRates", "BaseRates operations")
+	baseRates.POST("", formatDoc("Upload baseRates", ""), tonic.Handler(r.uploadBaseRateHandler, http.StatusCreated))
+	baseRates.GET("/:baseRate", formatDoc("Get BaseRate", ""), tonic.Handler(r.getBaseRateHandler, http.StatusOK))
+	baseRates.GET("", formatDoc("Get BaseRates by country", ""), tonic.Handler(r.getBaseRatesHandler, http.StatusOK))
+
+	packages := v1.Group("/packages", "Packages", "Packages operations")
+	packages.PUT("", formatDoc("Add Package", ""), tonic.Handler(r.AddPackageHandler, http.StatusCreated))
+	packages.GET("/:package", formatDoc("Get package", ""), tonic.Handler(r.getPackageHandler, http.StatusOK))
+	packages.PATCH("", formatDoc("Update Package", ""), tonic.Handler(r.UpdatePackageHandler, http.StatusOK))
```

Comment:
PATCH is used to "update-by-patch" an existing resource. So it should be used over an identifiable REST resource, clearly specified as a path parameter.